packages/fluentui/docs/src/prototypes/VirtualizedStickyTree/VirtualizedStickyTree.tsx

```diff
@@ -0,0 +1,468 @@
+
+        onActiveItemsIdChange([...newStickyTopIds, ...newStickyBottomIds], [itemProps.id]);
+        passOnStickyTopIds(newStickyTopIds);
+        passOnStickyBottomIds(newStickyBottomIds);
+      } else if (itemProps.expanded) {
+        // item is not sticky parent, and expanded. therefore collapse it
+        onActiveItemsIdChange([itemProps.id]);
+      } else {
+        // item is not sticky parent, and collapsed. therefore expand it
+
+        // find the sticky parent of this item
+        // IMPORTANT: this is why sticky tree supports only 2/3 levels
+        const toplvlParentId = itemProps.parent;
+        // here's the logic for finding sticky parent, if we want to support more levels:
+        // let parentItem = item;
+        // while (parentItem?.props?.parent) {
+        //   parentItem = renderedItems.find(
+        //     (item: React.ReactElement<TreeItemProps>) =>
+        //       item.props.id === parentItem.props.parent
+        //   );
+        // }
+        // const toplvlParentId = parentItem.props.id;
+
+        // stick all its sticky parent's right siblings to bottom
+        const newStickyBottomIds = toplvlItemsId.slice(toplvlItemsId.findIndex(id => id === toplvlParentId) + 1);
+        passOnStickyBottomIds(newStickyBottomIds);
+
+        onActiveItemsIdChange(newStickyBottomIds, [itemProps.id]);
+      }
+    },
+    [toplvlItemsId, onActiveItemsIdChange, passOnStickyTopIds, passOnStickyBottomIds, passItemIdTobeFocused],
+  );
+
+  const onClickSticky = React.useCallback(
+    (_event: React.SyntheticEvent<HTMLElement>, itemProps: TreeItemProps) => {
+      passItemIdTobeFocused(itemProps.id); // retain focus after re-render
+
+      if (itemProps.expanded) return; // it is already the main item stick to top, do nothing
+
+      // expand it and stick it to the last one on top
+      const toFocusItemIndex = toplvlItemsId.findIndex(id => id === itemProps.id);
+
+      const newStickyTopIds = toplvlItemsId.slice(0, toFocusItemIndex + 1); // every left siblings till itself
+      const newStickyBottomIds = toplvlItemsId.slice(toFocusItemIndex + 1); // every right siblings
+
+      onActiveItemsIdChange([...newStickyTopIds, ...newStickyBottomIds], [itemProps.id]);
+      passOnStickyTopIds(newStickyTopIds);
+      passOnStickyBottomIds(newStickyBottomIds);
+    },
+    [toplvlItemsId, onActiveItemsIdChange, passOnStickyTopIds, passOnStickyBottomIds, passItemIdTobeFocused],
+  );
+
+  const ref = React.useRef(null); // ref for entire list
+
+  // When using keyboard, and navigate to non-sticky items, they could be hidden behind sticky headers.
+  // Scroll to make the focused non-sticky items always visible
+  const handleOnFocus = React.useCallback(
+    (e: React.SyntheticEvent<HTMLElement>) => {
+      if (whatInput.ask() !== 'keyboard') return;
+
+      let itemRef = e.target as any;
+      if (itemRef.tagName !== 'DIV') itemRef = itemRef.parentNode; // when treeTitle focused, get its outer treeItem
```

Comment:
can you check for `treeTitleClassName` class presence instead of the tag name? tag names can be customized.

Reply:
You're right checking tagname is not good. I pushed a new change with this check:
```ts
if (itemRef.className.includes('ui-tree__title'))
```